src/GT.Core/Services/Impl/GTCompanyService.cs

```diff
@@ -92,14 +153,114 @@ public async Task<bool> ExistsByNameAsync(string name)
 			}
 		}
 
-		public Task<CompanyDTO> GetAsync()
+		public async Task<List<CompanyDTO>> GetAsync()
 		{
-			throw new NotImplementedException();
+			try
+			{
+				var entities = await _companyRepository.GetAll().ToListAsync();
```

Comment:
Shouldn't the resulting IQueryable from .GetAll() be projected with a .Select() call before calling .ToListAsync() to minimize the amount of data read from the database?

Reply:
Dont know whats best practice. I will need a walkthrough of this

Reply:
Of course, I can walk through it later today. In my opinion it would be fine to merge this in the interest of implementing the other changes first, and then fix it later today.

Reply:
OF course, I can walk through this later today when there is time. In my opinion it would be fine to merge this in the interest of implementing the other features, and then looking at this later today.